apps/dashr-studybrowser/app.R

```diff
@@ -0,0 +1,286 @@
+appName <- Sys.getenv("DASH_APP_NAME")
+pathPrefix <- sprintf("/%s/", appName)
+Sys.setenv(DASH_ROUTES_PATHNAME_PREFIX = pathPrefix, 
+           DASH_REQUESTS_PATHNAME_PREFIX = pathPrefix)
+setwd("app/apps/dashr-studybrowser")
+
+library(jsonlite)
+library(dashR)
+library(dashCoreComponents)
+library(dashHtmlComponents)
+library(data.table)
+library(stringr)
+library(magrittr)
+library(foreach)
+library(rlist)
+library(glue)
+library(plotly)
+
+
+
+#Syntactic sugar. just relabeling
+l. <- list
+#"Dict-List" of colors
+group_colors = l.('negative'='light blue', 'test'='green', 'positive'='red')
+
+
+returnOptions <- function(t_a, s_d){
+  studies <- (s_d[s_d$test_article==t_a])$study_id %>% unique(.)
+  foreach(i=studies) %do% l.('label'=glue('{t_a} (study: {i})'), 'value'=i)
+    
+
+}
+
+asterisks <- function(n){
+  if(0.1<=n){
+    return(glue('! \n '))
+  } else if(0.05<=n && n<0.1){
+    return(glue('. \n '))    
+  } else if(0.01<=n && n<0.05){
+    return(glue('*  \n '))
+  } else if(0.001<=n && n<0.01){
+    return(glue(' **  \n '))
+  } else{
+    return(glue('***  \n  '))
+  }
+  
+}
+
+default_study_data <- fread('study.csv')
+
+default_studyid <- "AS100"
+default_study_input <- default_study_data[study_id==default_studyid]
+
+
+default_test_articles <- (default_study_data[default_study_data$group_type=='test'] )$test_article %>% unique(.)
+default_options = foreach(i=default_test_articles, combine='c') %do% returnOptions(i, default_study_data)
```

Comment:
lapply works the best here with `base` -- `foreach` was used here with originally `%dopar%` instead of `%do%`, trying to improve the performance! I have a version using `lapply` in my local folder; can just push that :)